experiments/AtmosLES/stable_bl_kosovic.jl

```diff
@@ -0,0 +1,387 @@
+    e_pot = _grav * z
+    ρe_tot = ρ * total_energy(e_kin, e_pot, TS)
+
+    # Assign initial conditions for prognostic state variables
+    state.ρ = ρ
+    state.ρu = SVector(ρu, ρv, ρw)
+    state.ρe = ρe_tot
+    state.moisture.ρq_tot = ρ * q_tot
+
+    if z <= FT(50) # Add random perturbations to bottom 50m of model
+        state.ρe += rand() * ρe_tot / 100
+    end
+end
+
+function surface_temperature_variation(state, aux, t)
+    FT = eltype(state)
+    ρ = state.ρ
+    q_tot = state.moisture.ρq_tot / ρ
+    θ_liq_sfc = FT(265) - FT(1 / 4) * (t / 3600)
+    TS = LiquidIcePotTempSHumEquil(param_set, θ_liq_sfc, ρ, q_tot)
+    return air_temperature(TS)
+end
+
+function config_problem(FT, N, resolution, xmax, ymax, zmax)
+
+    ics = init_problem!     # Initial conditions
+
+    C_smag = FT(0.23)     # Smagorinsky coefficient
+    C_drag = FT(0.001)    # Momentum exchange coefficient
+    z_sponge = FT(300)     # Start of sponge layer
+    α_max = FT(0.75)       # Strength of sponge layer (timescale)
+    γ = 2                  # Strength of sponge layer (exponent)
+    u_geostrophic = FT(8)        # Eastward relaxation speed
+    u_slope = FT(0)              # Slope of altitude-dependent relaxation speed
+    v_geostrophic = FT(0)        # Northward relaxation speed
+    f_coriolis = FT(1.39e-4) # Coriolis parameter
+    q_sfc = FT(0)
+    u_star = FT(0.30)
+
+    # Assemble source components
+    source = (
+        Gravity(),
+        StableBLSponge{FT}(
+            zmax,
+            z_sponge,
+            α_max,
+            γ,
+            u_geostrophic,
+            u_slope,
+            v_geostrophic,
+        ),
+        StableBLGeostrophic{FT}(
+            f_coriolis,
+            u_geostrophic,
+            u_slope,
+            v_geostrophic,
+        ),
+    )
+
+    # Choose default IMEX solver
+    ode_solver_type = ClimateMachine.ExplicitSolverType()
+
+    moisture_flux = FT(0)
+    # Assemble model components
+    model = AtmosModel{FT}(
+        AtmosLESConfigType,
+        param_set;
+        turbulence = SmagorinskyLilly{FT}(C_smag),
+        moisture = EquilMoist{FT}(; maxiter = 5, tolerance = FT(0.1)),
+        source = source,
+        boundarycondition = (
+            AtmosBC(
+                momentum = Impenetrable(DragLaw(
+                    # normPu_int is the internal horizontal speed
+                    # P represents the projection onto the horizontal
+                    (state, aux, t, normPu_int) -> (u_star / normPu_int)^2,
+                )),
+                energy = BulkFormulaEnergy(
+                    (state, aux, t, normPu_int) -> C_drag,
+                    (state, aux, t) ->
+                        surface_temperature_variation(state, aux, t),
+                    (state, aux, t) -> q_sfc,
+                ),
+                moisture = BulkFormulaMoisture(
+                    (state, aux, t, normPu_int) -> C_drag,
+                    (state, aux, t) -> q_sfc,
```

Comment:
@szy21: if we had changed `bc_moisture.jl` also in #1454, then we would have needed to call `surface_temperature_variation()` even though the temperature is not required in the moisture flux computation, right?